File: src/lib/scala/Slate.Common/src/main/scala/slate/common/databases/DbType.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2016 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */
package slate.common.databases


abstract class DbType(val name:String, val driver:String)

case object DbTypeMySql     extends DbType("mysql"    , "com.mysql.jdbc.Driver")
case object DbTypeSqlServer extends DbType("sqlserver", "com.microsoft.sqlserver.jdbc.SQLServerDriver")
case object DbTypeMemory    extends DbType("memory"   , "com.slatekit.entities.repository-in-memory")

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/db/DbType.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */
package slatekit.common.db

sealed class DbType(val name: String, val driver: String) {
  object DbTypeMySql  : DbType("mysql", "com.mysql.jdbc.Driver")
  object DbTypePGres  : DbType("pgres", "org.postgresql.Driver")
  object DbTypeSqLite : DbType("sqlite", "org.sqlite.JDBC")
  object DbTypeMemory : DbType("memory", "com.slatekit.entities.repository-in-memory")
}
